### PR TITLE
refactor(yaml): add `sanitizeInput()` function

### PR DESCRIPTION
**Pull request by timreichen**
- moves code into a dedicated `sanitizeInput()` function

### CODE REVIEW

**Comment by kt3k** [comment]
`sanitizeInput` is only used once. I don't see this change reasonable.

**Comment by iuioiua** [comment]
I don't see the purpose of this change either.

**Comment by timreichen** [comment]
> `sanitizeInput` is only used once. I don't see this change reasonable.

I will create a second PR that uses generator functions for documents in `loader.ts`. Therefore `load()` and `loadDocuments()` will need to call `sanitizeInput()` individually. The gain will be to throw early if there are more than one document in `load()` instead of parsing all documents and then throw.

File: yaml/_loader/loader.ts [hunk 1]
```diff
@@ -1690,10 +1687,21 @@ export function loadDocuments(
...
+  return input;
+}
+
+export function loadDocuments(
+  input: string,
+  options?: LoaderStateOptions,
+): unknown[] {
+  input = sanitizeInput(input);
+
+  options = options || {};
```
Comment (iuioiua) [inline yaml/_loader/loader.ts:1702]:
```suggestion
export function loadDocuments(
  input: string,
  options: LoaderStateOptions = {},
): unknown[] {
  input = sanitizeInput(input);
```

**Review by iuioiua — CHANGES_REQUESTED** [review]

**Review by kt3k — APPROVED** [review]
LGTM

**Review by iuioiua — APPROVED** [review]

**Pull request merged**